query-connector/src/app/queryBuilding/dataState/utils.tsx

```diff
@@ -0,0 +1,154 @@
+ * @param setQueries - Function to update the state of queries.
+ * @param context - The data context used to update shared state.
+ */
+export const handleDelete = async (
+  queryName: string,
+  queryId: string,
+  queries: CustomUserQuery[],
+  setQueries: React.Dispatch<React.SetStateAction<CustomUserQuery[]>>,
+  context: DataContextValue | undefined,
+) => {
+  const result = await deleteQueryById(queryId);
+  if (result.success) {
+    toast.error(`${queryName} (${queryId}) has been deleted.`, {
+      autoClose: 2000,
+    });
+    const updatedQueries = queries.filter(
+      (query) => query.query_id !== queryId,
+    );
+    setQueries(updatedQueries);
+
+    if (context) {
+      context.setData(updatedQueries);
+    }
+  } else {
+    console.error(result.error);
+  }
+};
+
+/**
+ * Confirms the deletion of a user query by toggling a modal.
+ * @param queryName - The name of the query to confirm deletion for.
+ * @param queryId - The ID of the query to confirm deletion for.
+ * @param setSelectedQuery - Function to set the currently selected query for deletion.
+ * @param modalRef - Reference to the modal component.
+ */
+export const confirmDelete = (
+  queryName: string,
+  queryId: string,
+  setSelectedQuery: React.Dispatch<
+    React.SetStateAction<{ queryName: string; queryId: string } | null>
+  >,
+  modalRef: RefObject<ModalRef>,
+) => {
+  setSelectedQuery({ queryName, queryId });
+  modalRef.current?.toggleModal();
+};
+
+/**
+ * Copies the query ID to the clipboard.
+ * @param queryName - The name of the query to copy the ID for.
+ * @param queryId - The ID of the query to copy.
+ */
+export const handleCopy = (queryName: string, queryId: string) => {
+  navigator.clipboard
+    .writeText(queryId)
+    .then(() => {
+      toast.success(`${queryName} (${queryId}) copied successfully!`, {
+        autoClose: 2000,
+      });
+    })
+    .catch((error) => {
+      console.error("Failed to copy text:", error);
+    });
+};
+
+/**
+ * Handles the creation of a new query by redirecting to the query building page.
+ * @param router - Next.js router for navigation.
+ * @param setLoading - Function to set the loading state.
+ */
+export const handleClick = async (
+  router: ReturnType<typeof useRouter>,
+  setLoading: React.Dispatch<React.SetStateAction<boolean>>,
+) => {
+  setLoading(true);
+
+  // Redirect to query updating/editing page
+  router.push("/queryBuilding/buildFromTemplates");
+};
+
+/**
+ * Renders a modal to confirm the deletion of a user query.
+ * @param modalRef - Reference to the modal component.
+ * @param selectedQuery - The currently selected query for deletion.
+ * @param handleDelete - Function to handle the deletion workflow.
+ * @param queries - The current list of user queries.
+ * @param setQueries - Function to update the state of queries.
+ * @param context - The data context used to update shared state.
+ * @returns The JSX element for the modal.
+ */
+export const renderModal = (
```

Comment:
could we move this code into the `Modal.tsx` file (maybe renaming this method to be `renderDeletionModal` / tweaking the inputs to be a bit more generic) then export it to be consumed in the places where we need modals to be rendered? 

That way, the UI code that controls the look and styling of our modals are centralized and we can share this work further down the line in other places where we need modals

Reply:
makes sense. I moved `Modal` and `DeleteModal` into a new folder. lmk your thoughts!